### ISSUE TITLE
This application practically never works on Galaxy S9 FE

### ISSUE THREAD

**Issue by wlinna** [ray-pH/Nabla-TypeMath-android#11]
Hello,
I just installed the software on my tablet. I gave Nabla accessibility permissions and added it to battery saving exceptions. Nabla says the service is on. The problem is that 99 % of the time the application doesn't work at all.

For example, if I input `.alpha.`, I might get the desired unicode character, but most of the time I will not. I have never been able to make something like  .x^2. work, and .x_2. has worked only twice. I have tried with both a physical keyboard and the touch screen. I use Gboard by the way. I have tried at least two different programs but neither of them work consistently with Nabla.

I installed Nabla from Play, and I'm running version 1.3.4 on Android 14 One UI 6.0.

**Comment by ray-pH** [comment]
Yeah, unfortunately, this is a common issue. The app is just broken on some phones, while it works just fine on others. 

unfortunately, I don't have easy access to those phones where I can test and investigate something.  I'll see if I can try to get my hands on those phones/emulator. But It might be a while

**Comment by ray-pH** [comment]
Wait a minute, just to make sure, did you turn off the `Quick Mode`?
Because since 1.2, the default setting is with the `Quick Mode` turned on.
In that mode, you don't write `.alpha.<space>` but `.alpha<space>`

**Comment by wlinna** [comment]
Thank you for your responses.
I tested with both Quick Mode turned on and off. Sometimes Quick Mode worked, sometimes "Slow" Mode worked. I wasn't able to establish any pattern, unfortunately.

**Comment by ray-pH** [comment]
Okay, thanks for the info

**Comment by wlinna** [comment]
Just tested with Samsung Keyboard, but it didn't work any better, unfortunately

**Comment by ray-pH** [comment]
![image](https://github.com/ray-pH/Nabla-TypeMath-android/assets/57003984/e0d098c2-2bf8-4d6d-8d5b-c3a1c137f1a2)
![image](https://github.com/ray-pH/Nabla-TypeMath-android/assets/57003984/6fc15f9d-e7a1-40a2-b686-9238a631dcb2)


I tried running the app in the Samsung Remote Test Lab, and I didn't experience any issue

**Comment by ray-pH** [comment]
Are you comfortable with running a debug version of the apk?

I'm interested in seeing the debug message like these
![image](https://github.com/ray-pH/Nabla-TypeMath-android/assets/57003984/459c5db4-7716-4f9a-bf67-46787a083e16)

If not, that's okay.

**Comment by wlinna** [comment]
I guess I could risk trusting you :P . How would I get the debug messages out?

**Comment by ray-pH** [comment]
I just remembered that on the early version of the app, there's a weird quirk where if you have [Typing Hero](https://play.google.com/store/apps/details?id=sen.typinghero&hl=en&gl=US) installed and enabled, it'll help Nabla works. I think that quirk should be gone in the later version, but maybe you can try that first.

![image](https://github.com/ray-pH/Nabla-TypeMath-android/assets/57003984/656b7733-b7ee-480f-a574-e61e783255f3)

**Comment by wlinna** [comment]
I'm not very comfortable installing proprietary software that has access to my keystrokes across the system, but thanks for the recommendation :)

**Comment by ray-pH** [comment]
> I guess I could do it . How would I get the debug messages out?

I think the only way is via ADB. Are you familiar with that?
I tried checking some logcat apps, but none of them seems to pick up the log from an AccessibilityService

**Comment by ray-pH** [comment]
Does the app stop working immediately, or does it take a while until the service stops working?

**Comment by wlinna** [comment]
> > I guess I could do it . How would I get the debug messages out?
> 
> I think the only way is via ADB. Are you familiar with that? I tried checking some logcat apps, but none of them seems to pick up the log from an AccessibilityService

Not very familiar, but I don't suppose it will be too hard. I might try that on Sunday

**Comment by ray-pH** [comment]
I've been able to recreate the issue. It's somehow app-specific. The issue started happening when I used Chrome (weirdly enough chrome in my phone works fine, that issue only occurs in Samsung), and maybe I'll check Joplin next.

I think I can investigate myself from there

**Comment by wlinna** [comment]
It wasn't app specific for me originally. However, I tried this again recently and didn't notice any problem anymore.
